Update Adding Devices doc for device repo (#380)
TheThingsIndustries/lorawan-stack-docs: doc/content/devices/adding-devices/_index.md

## Adding Devices using the Console

> Devices are managed under applications. An application can contain an unlimited number of devices, but it can be helpful to sort devices in to applications by function or geographical area, to make the integrations and live data views more useful.

Reviewer: This should be moved up (above this title) because it refers to devices and applications in general

To create a device, first open the application you wish to add the device in. Go to **End devices** in the left menu and click on **+ Add end device** to reach the end device registration page.

{{< figure src="application-overview.png" alt="Application overview" >}}

### Using the LoRaWAN Device Repository

The [LoRaWAN device repository](https://github.com/TheThingsNetwork/lorawan-devices) contains device profiles, LoRaWAN information, codecs, and more, for many LoRaWAN devices. Using the device repository to add devices in {{% tts %}} automatically uses the correct LoRaWAN version and regional parameters version, which means less information for you to find!

> To use the device repository, make sure the **From the LoRaWAN Device Repository** button is selected. Then, select the **Brand**, **Model**, **Hardware Version**, **Software Version**, and **Region** for your device.

Reviewer: ```suggestion
To use the device repository, make sure the **From the LoRaWAN Device Repository** tab is selected. Then, select the **Brand**, **Model**, **Hardware Version**, **Software Version**, and **Region** for your device.
```

{{< note "If your device is not in the device repository, see [Manually Create End Device](#manually-create-end-device) below." />}}

{{< figure src="device-repo.png" alt="Creating a new device with the Device Repository" >}}

Choose a **Frequency plan** appropriate for your region. Your device and gateway must use the same frequency plan to communicate.

Enter a **JoinEUI/AppEUI** if provided by your manufacturer. If your device is programmable, you may use all 0's, and then program the same JoinEUI/AppEUI in the device.

> Enter your **DevEUI**.

Reviewer: Maybe add a similar line as for the JoinEUI/AppEUI, because DevEUI can be provided by the manufacturer, but I think we can issue DevEUIs now too

Author: I don't know what the official word is on this, so going to leave it for now.

If your manufacturer provides an **AppKey**, enter it. Otherwise, use the generate button to create one, and program it in to your device.

Finally, give your device a unique **End device ID**, and click the **Register end device** button to create the end device.

{{< note >}}
See [ID and EUI constraints]({{< ref "reference/id-eui-constraints" >}}) for guidelines about choosing a unique ID.
{{</ note >}}

{{< figure src="device-repo-settings.png" alt="Device information" >}}

The device is now activated, and will appear as connected in {{% tts %}} once it sends an uplink.

### Manually Create End Device

If your device is not in the device repository, you may manually register it.

> Make sure the **Manually** button is selected.

Reviewer: ```suggestion
Make sure the **Manually** tab is selected.
```

> {{< figure src="register-manually.png" alt="Register manually" >}}

Reviewer: This image is redundant and too enlarged

Choose the **Activation Mode**.

Select the device **LoRaWAN version**.

If using Over-the-Air-Activation (OTAA) and an External Join Server, tick the **External Join Server** button.

> The **Network Server**, **Application Server**, and **Join Server** addresses should correctly point to the address of your {{% tts %}} deployment.

Reviewer: ```suggestion
The **Network Server**, **Application Server**, and **Join Server** addresses should correctly point to the address of {{% tts %}} deployment you are using.
```

Click the **Start** button to proceed to the Basic Settings page.

{{< figure src="manual-settings.png" alt="Register manually" >}}

{{< note >}}

> This guide covers adding a device using [OTAA]({{< ref "reference/glossary#over-the-air-activation" >}}) (the most secure and preferred activation method) and [LoRaWAN version]({{< ref "reference/glossary#over-the-air-activation" >}}) MAC V1.0.2 (the most common LoRaWAN version, although newer version are better and more secure). Names and keys may vary slightly for other versions.

Reviewer: Bad link for LoRaWAN version

> This guide covers adding a device using [OTAA]({{< ref "reference/glossary#over-the-air-activation" >}}) (the most secure and preferred activation method) and [LoRaWAN version]({{< ref "reference/glossary#over-the-air-activation" >}}) MAC V1.0.2 (the most common LoRaWAN version, although newer version are better and more secure). Names and keys may vary slightly for other versions.

Reviewer: ```suggestion
This guide covers adding a device using [OTAA]({{< ref "reference/glossary#over-the-air-activation" >}}) (the most secure and preferred activation method) and [LoRaWAN version]({{< ref "reference/glossary#over-the-air-activation" >}}) MAC V1.0.2 (the most common LoRaWAN version, although newer versions are better and more secure). Names and keys may vary slightly for other versions.
```
